diff --git a/applications/connectivity_bridge/boards/thingy91x_nrf5340_cpuapp.conf b/applications/connectivity_bridge/boards/thingy91x_nrf5340_cpuapp.conf
index c59f3dc494..da01110297 100644
--- a/applications/connectivity_bridge/boards/thingy91x_nrf5340_cpuapp.conf
+++ b/applications/connectivity_bridge/boards/thingy91x_nrf5340_cpuapp.conf
@@ -34,3 +34,7 @@ CONFIG_REBOOT=y
 CONFIG_GPIO=y
 
 CONFIG_DK_LIBRARY=y
+
+# Bootloader firmware information
+CONFIG_FW_INFO=y
+CONFIG_SECURE_BOOT_STORAGE=y
diff --git a/applications/connectivity_bridge/sample.yaml b/applications/connectivity_bridge/sample.yaml
index 3074559e7f..06d3a19a72 100644
--- a/applications/connectivity_bridge/sample.yaml
+++ b/applications/connectivity_bridge/sample.yaml
@@ -12,3 +12,14 @@ tests:
       - thingy91/nrf52840
       - thingy91x/nrf5340/cpuapp
     tags: ci_build sysbuild
+  applications.connectivity_bridge.bootloader_update:
+    build_only: true
+    sysbuild: true
+    integration_platforms:
+      - thingy91x/nrf5340/cpuapp
+    platform_allow:
+      - thingy91x/nrf5340/cpuapp
+    tags: ci_build sysbuild
+    extra_args:
+      - mcuboot_CONFIG_FW_INFO_FIRMWARE_VERSION=2
+      - CONFIG_BUILD_S1_VARIANT=y
diff --git a/applications/connectivity_bridge/src/modules/usb_bulk_commands.c b/applications/connectivity_bridge/src/modules/usb_bulk_commands.c
index 55367dc403..97653fbc26 100644
--- a/applications/connectivity_bridge/src/modules/usb_bulk_commands.c
+++ b/applications/connectivity_bridge/src/modules/usb_bulk_commands.c
@@ -8,6 +8,9 @@
 #include <zephyr/sys/reboot.h>
 #include <zephyr/drivers/gpio.h>
 #include <dk_buttons_and_leds.h>
+#include <fw_info.h>
+#include <bl_storage.h>
+#include <ncs_commit.h>
 
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(bulk_commands, CONFIG_BRIDGE_BULK_LOG_LEVEL);
@@ -16,10 +19,12 @@ LOG_MODULE_REGISTER(bulk_commands, CONFIG_BRIDGE_BULK_LOG_LEVEL);
 #define ID_DAP_VENDOR15			(0x80 + 15)
 #define ID_DAP_VENDOR16			(0x80 + 16)
 #define ID_DAP_VENDOR17			(0x80 + 17)
+#define ID_DAP_VENDOR18			(0x80 + 18)
 #define ID_DAP_VENDOR_NRF53_BOOTLOADER	ID_DAP_VENDOR14
 #define ID_DAP_VENDOR_NRF91_BOOTLOADER	ID_DAP_VENDOR15
 #define ID_DAP_VENDOR_NRF53_RESET	ID_DAP_VENDOR16
 #define ID_DAP_VENDOR_NRF91_RESET	ID_DAP_VENDOR17
+#define ID_DAP_VENDOR_NRF53_VERSION	ID_DAP_VENDOR18
 
 #define DAP_COMMAND_SUCCESS		0x00
 #define DAP_COMMAND_FAILED		0xFF
@@ -125,6 +130,22 @@ size_t dap_execute_cmd(uint8_t *in, uint8_t *out)
 		}
 		goto success;
 	}
+	if (in[0] == ID_DAP_VENDOR_NRF53_VERSION) {
+		const struct fw_info *s0_info = fw_info_find(s0_address_read());
+		const struct fw_info *s1_info = fw_info_find(s1_address_read());
+		__ASSERT_NO_MSG(s0_info != NULL);
+		__ASSERT_NO_MSG(s1_info != NULL);
+		int len = snprintf(out + 2, 64 - 2,
+				   "S0: %u, S1: %u, app: " NCS_COMMIT_STRING,
+				   s0_info->version, s1_info->version);
+		if (len < 0) {
+			LOG_ERR("Failed to format version string");
+			goto error;
+		}
+		out[0] = in[0];
+		out[1] = len;
+		return len + 2;
+	}
 
 error:
 	/* default reply: command failed */
diff --git a/applications/connectivity_bridge/sysbuild/mcuboot/boards/thingy91x_nrf5340_cpuapp.conf b/applications/connectivity_bridge/sysbuild/mcuboot/boards/thingy91x_nrf5340_cpuapp.conf
index 37c7e95b15..8bbd4ddf94 100644
--- a/applications/connectivity_bridge/sysbuild/mcuboot/boards/thingy91x_nrf5340_cpuapp.conf
+++ b/applications/connectivity_bridge/sysbuild/mcuboot/boards/thingy91x_nrf5340_cpuapp.conf
@@ -58,3 +58,5 @@ CONFIG_BOOT_SERIAL_IMG_GRP_IMAGE_STATE=y
 
 # Skip checks on the secondary image to make it possible to update MCUBoot on S1/S0
 CONFIG_MCUBOOT_VERIFY_IMG_ADDRESS=n
+
+CONFIG_BOOT_SERIAL_NO_APPLICATION=y
