diff --git a/nrf_wifi/hw_if/hal/src/hal_api.c b/nrf_wifi/hw_if/hal/src/hal_api.c
index 1b23fa42..5765e383 100644
--- a/nrf_wifi/hw_if/hal/src/hal_api.c
+++ b/nrf_wifi/hw_if/hal/src/hal_api.c
@@ -1258,6 +1258,7 @@ void nrf_wifi_hal_dev_rem(struct nrf_wifi_hal_dev_ctx *hal_dev_ctx)
 
 	nrf_wifi_osal_spinlock_free(hal_dev_ctx->lock_hal);
 	nrf_wifi_osal_spinlock_free(hal_dev_ctx->lock_rx);
+	nrf_wifi_osal_spinlock_free(hal_dev_ctx->lock_recovery);
 
 	nrf_wifi_utils_q_free(hal_dev_ctx->event_q);
 
